src/metabase/automagic_dashboards/core.clj

```diff
@@ -713,10 +719,13 @@
 (defmethod inject-root Field
   [context field]
   (let [field (assoc field
-                :link   (->> context
-                             :tables
-                             (m/find-first (comp #{(:table_id field)} u/the-id))
-                             :link)
+                :link (->> context
```

Comment:
Is it possible to add a helpful error in the case it otherwise blows up? Doesn't improve the functionality, but helps in understanding. An error like "Unable to find Table to link root" or something versus "integer is not seqable" or whatever the blow up is here.

Reply:
Yeah, just need to remember how I reproduced this. It was kind of a corner case. Part of the next several steps is putting together more complicated cases to explicitly unearth and test this stuff.